tests/e2e/Services/VCS/VCSConsoleClientTest.php

```diff
@@ -0,0 +1,316 @@
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId,
+            'providerRepositoryId' => 'randomRepositoryId'
+        ]);
+
+        $this->assertEquals(404, $repositoryBranches['headers']['status-code']);
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testCreateFunctionUsingVCS(string $installationId)
+    {
+        $function = $this->client->call(Client::METHOD_POST, '/functions', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'functionId' => ID::unique(),
+            'name' => 'Test',
+            'execute' => [Role::user($this->getUser()['$id'])->toString()],
+            'runtime' => 'php-8.0',
+            'entrypoint' => 'index.php',
+            'events' => [
+                'users.*.create',
+                'users.*.delete',
+            ],
+            'schedule' => '0 0 1 1 *',
+            'timeout' => 10,
+            'installationId' => $installationId,
+            'providerRepositoryId' => $this->providerRepositoryId,
+            'providerBranch' => 'main',
+        ]);
+
+        $this->assertEquals(201, $function['headers']['status-code']);
+
+        return [
+            'installationId' => $installationId,
+            'functionId' => $function['body']['$id']
+        ];
+    }
+
+    /**
+     * @depends testCreateFunctionUsingVCS
+     */
+    public function testUpdateFunctionUsingVCS(array $data, string $providerRepositoryId2 = '705764449')
+    {
+        $function = $this->client->call(Client::METHOD_PUT, '/functions/' . $data['functionId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'functionId' => ID::unique(),
+            'name' => 'Test',
+            'execute' => [Role::user($this->getUser()['$id'])->toString()],
+            'runtime' => 'php-8.0',
+            'entrypoint' => 'index.php',
+            'events' => [
+                'users.*.create',
+                'users.*.delete',
+            ],
+            'schedule' => '0 0 1 1 *',
+            'timeout' => 10,
+            'installationId' => $data['installationId'],
+            'providerRepositoryId' => $providerRepositoryId2,
+            'providerBranch' => 'main',
+        ]);
+
+        $this->assertEquals(200, $function['headers']['status-code']);
+
+        return $function['body']['$id'];
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testCreateRepository(string $installationId)
```

Comment:
Let's include failure test. When you try to create repository twice, second one should fail as it will be name conflict.